services/src/modules/resource-repository/types.ts

```diff
@@ -3,6 +3,7 @@ export interface ResourceGroup {
     upstreams: Upstream[];
     upstreamClientCredentials: UpstreamClientCredentials[];
     policies: Policy[];
+    policyAttachments?: PolicyAttachments;
```

Comment:
Leave a comment about this being a "runtime-only" property? Since the rest are assumed to always exist

Reply:
Come to think of it, it will be populated on the first call to `fetchLatest` same as the others, and will always exist (empty object if there are no attachments at all, same as the other fields here).
All of them are actually `runtime-only` properties representing what is stored in the underlying repository, with the only different between the attachments and the other properties being that the other properties are all stored in one file and the attachments are stored in other files.

However, when using this interface we don't really care that it is stored in other files than the other properties, so I'm actually leaning towards removing the `?` here and treating it like any other property.

One key difference is that the contents of this property are calculated based on the `policies` property and not created/updated directly by a user, but it only happens once when we save the policy and then saved this way in the repository, so I would still say we should treat it like the other properties here (but I will add a comment to highlight this difference)